src/unix/udp.c

```diff
@@ -663,6 +755,46 @@ int uv_udp_set_membership(uv_udp_t* handle,
   }
 }
 
+
+int uv_udp_set_source_membership(uv_udp_t* handle,
+                                 const char* multicast_addr,
+                                 const char* interface_addr,
+                                 const char* source_addr,
+                                 uv_membership membership) {
+  int err;
+  union sockaddr {
```

Comment:
why not use `struct sockadr_storage` here?

Reply:
Yes, I thought about it when going through the changes but went for this approach for a couple of reasons:
- I didn't want to change the original PR much.
- The [current multicast method](http://docs.libuv.org/en/v1.x/udp.html#c.uv_udp_set_membership) already uses strings instead of `sockaddr_`  structs (which I guess that's the reason why this interface was chosen).

That said, I think it makes more sense using `sockaddr_`  structs in both methods, but I guess that'll have to go into `master`?

Reply:
@saghul disregard my last comment. I though you were talking about the function arguments :man_facepalming: 

About using `struct sockaddr_storage` I [already commented about it in the original PR](https://github.com/libuv/libuv/pull/964#discussion_r166565124), but thought this option looked cleaner.

Reply:
I see. Not my cup of tea, but I won't block this based on that.